Utils/WorldZone.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using CitizenFX.Core;
+using CitizenFX.Core.Native;
+
+namespace FivePD-API.Utils
+{
+    internal class WorldZone
+    {
+        private static readonly List<WorldZone> Zones = new List<WorldZone>();
+        private readonly EWorldZoneCounty county;
+
+        private string GameName { get; }
+
+        internal string RealAreaName { get; }
+
+        internal string County
+        {
+            get
+            {
+                var r = new Regex(@"
+                (?<=[A-Z])(?=[A-Z][a-z]) |
+                 (?<=[^A-Z])(?=[A-Z]) |
+                 (?<=[A-Za-z])(?=[^A-Za-z])", RegexOptions.IgnorePatternWhitespace);
+
+                return r.Replace(county.ToString(), " ");
+            }
+        }
+
+        private WorldZone(string gameName, string areaName, EWorldZoneCounty county)
+        {
+            GameName = gameName;
+            RealAreaName = areaName;
+            this.county = county;
+        }
+
+
+        internal static WorldZone GetZoneAtPosition(Vector3 position)
+        {
+            if (Zones.Count <= 0)
+                RegisterWorldZones();

```

Comment:
Why is this not initialised in the constructor?

Reply:
No need to add it to the c.tor, saves populating a list without reason. If this is never used by the dev making use of the API then I see no reason to populate a list. Rather populate it on it's first call. This is also a static call, the c.tor creates each of the zones not the list.

Reply:
Per C# language specification, a static field is being populated prior to using the class for the first time automatically on an implementation-level, therefore making this unnecessary. If someone is never using/referencing this class, the static field will never be populated. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/variables#static-variables

Reply:
I would just initialise the variable inline and let C# decide when to allocate memory.